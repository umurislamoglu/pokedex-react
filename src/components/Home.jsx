import React from 'react'

const Home = () => {
    return (
        <div className=" d-flex flex-column align-items-center" style={{minHeight:"100vh", background:"url(https://d.wattpad.com/story_parts/799408952/images/15d1e58d8c7420fd393949237219.jpg)" , backgroundRepeat:"no-repeat",backgroundAttachment:"fixed",backgroundPosition:"center"}} >
 <main className="px-3 container">
    <h1 className="text-light mt-5">Pokedex</h1>
    <p className="lead text-light">The Pokédex (ポケモン図鑑; Pokemon Zukan; lit. "Illustrated Pokémon Encyclopedia") is an electronic device designed to catalogue and provide information regarding the various species of Pokémon featured in the Pokémon video game, anime and manga series. The name Pokédex is a neologism including "Pokémon" (which itself is a portmanteau of "pocket" and "monster") and "index". The Japanese name is simply "Pokémon Encyclopedia", as it can feature every Pokémon on it, depending on the Pokédex.</p>
    <p className="lead">
    </p>
  </main>

  <footer className="mt-auto text-white-50">
    <p>Pokemon Project for Ekinoks Software, by Umur İslamoğlu</p>
  </footer>
        </div>
    )
}

export default Home
