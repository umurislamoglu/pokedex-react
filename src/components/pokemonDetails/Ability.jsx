import React from 'react'

const Ability = ({ability}) => {
    return (
        <li data-testid="ability" className="list-group-item list-group-item-success text-capitalize fs-5 mt-2 rounded-pill border d-flex justify-content-center">
            {ability}
        </li>
    )
}

export default Ability
