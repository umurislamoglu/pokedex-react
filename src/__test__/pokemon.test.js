import React from "react";
import  ReactDOM  from "react-dom";
import Home from "../components/Home";
import Ability from './../components/pokemonDetails/Ability';
import {render , cleanup} from "@testing-library/react"
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer"


afterEach(cleanup)

it("renders without crashing", ()=>{
    const div =document.createElement("div");
    ReactDOM.render(<Home></Home>,div)
})

it("renders Ability correctly",()=>{
   const  {getByTestId} = render(<Ability ability = "Overgrow"></Ability>)
   expect(getByTestId("ability")).toHaveTextContent("Overgrow")
})

it("renders Ability correctly",()=>{
    const  {getByTestId} = render(<Ability ability = "Chlorophyll"></Ability>)
    expect(getByTestId("ability")).toHaveTextContent("Chlorophyll")
 })

it("matches snapshot", ()=> {
   const tree = renderer.create(<Ability ability = "Chlorophyll"></Ability>).toJSON()
   expect(tree).toMatchSnapshot();
}) 