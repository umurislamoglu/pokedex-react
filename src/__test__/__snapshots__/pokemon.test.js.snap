// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<li
  className="list-group-item list-group-item-success text-capitalize fs-5 mt-2 rounded-pill border d-flex justify-content-center"
  data-testid="ability"
>
  Chlorophyll
</li>
`;
